server/lib/tuist_web/controllers/api/oidc_controller.ex

```diff
@@ -0,0 +1,154 @@
+defmodule TuistWeb.API.OIDCController do
+  @moduledoc """
+  Controller for OIDC token exchange.
+
+  This controller handles the exchange of CI provider OIDC tokens for
+  short-lived Tuist access tokens. Currently supports GitHub Actions.
+  """
+
+  use OpenApiSpex.ControllerSpecs
+  use TuistWeb, :controller
+
+  alias OpenApiSpex.Schema
+  alias Tuist.Guardian
+  alias Tuist.OIDC
+  alias Tuist.Projects
+  alias TuistWeb.API.Schemas.Error
+
+  plug(
+    OpenApiSpex.Plug.CastAndValidate,
+    json_render_error_v2: true,
+    render_error: TuistWeb.RenderAPIErrorPlug
+  )
+
+  tags(["OIDC Authentication"])
+
+  @default_ci_scopes [
+    "project:cache:write",
+    "project:previews:write",
+    "project:bundles:write",
+    "project:tests:write",
+    "project:builds:write"
+  ]
+
+  @token_ttl_seconds 3600
+
+  operation(:exchange_token,
+    summary: "Exchange a CI provider OIDC token for a Tuist access token.",
```

Comment:
Is this specific to CI providers? If not, I'd drop CI from there and just say that we exchange an OIDC from a supported partner (not sure if partner is the right word here).

Reply:
yes, it is specific to CI providers. We don't support non-CI providers and we have no plans to do so. We can evolve the summary if that happens.